packages/patrol_cli/lib/src/coverage/coverage_tool.dart

```diff
@@ -0,0 +1,213 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io' as io;
+
+import 'package:adb/adb.dart';
+import 'package:coverage/coverage.dart' as coverage;
+import 'package:dispose_scope/dispose_scope.dart';
+import 'package:file/file.dart';
+import 'package:glob/glob.dart';
+import 'package:patrol_cli/src/base/logger.dart';
+import 'package:patrol_cli/src/coverage/device_to_host_port_transformer.dart';
+import 'package:patrol_cli/src/coverage/vm_connection_details.dart';
+import 'package:patrol_cli/src/devices.dart';
+import 'package:platform/platform.dart';
+import 'package:process/process.dart';
+import 'package:vm_service/vm_service_io.dart';
+
+class CoverageTool {
+  CoverageTool({
+    required FileSystem fs,
+    required ProcessManager processManager,
+    required Platform platform,
+    required Adb adb,
+    required DisposeScope parentDisposeScope,
+  })  : _fs = fs,
+        _processManager = processManager,
+        _platform = platform,
+        _adb = adb,
+        _disposeScope = DisposeScope() {
+    _disposeScope.disposedBy(parentDisposeScope);
+  }
+
+  final FileSystem _fs;
+  final ProcessManager _processManager;
+  final Platform _platform;
+  final Adb _adb;
+  final DisposeScope _disposeScope;
+
+  Future<void> run({
+    required String flutterPackageName,
+    required TargetPlatform platform,
+    required Logger logger,
+    required Set<Glob> ignoreGlobs,
+  }) async {
+    final homeDirectory =
+        _platform.environment['HOME'] ?? _platform.environment['USERPROFILE'];
+    final hitMap = <String, coverage.HitMap>{};
+
+    await _disposeScope.run(
+      (scope) async {
+        final logsProcess = await _processManager.start(
+          ['flutter', 'logs'],
```

Comment:
Pass `FlutterCommand` here, instead of calling hardcoded `flutter`.

Reply:
This is a special case here. The command is ran in home directory to avoid any project related errors like the one below

```
The Xcode project defines schemes: dev, prod
You must specify a --flavor option to select one of the available schemes.
```

If it's run in the home dir, does it make sense to run it using fvm (or are there any other use cases for setting flutter command)?

Reply:
Then I'm not sure if that's good approach. Globally set flutter version (using fvm / the standard way) may be different from the one set in the project. Isn't there a risk that this will fail due to versions incompatibility?

Reply:
`flutter logs` seems to just listen to adb on android, and on iOS it connects to the VM and listens to stdout. It seems unlikely to me that flutter changes so much that it fails.
The only other solution I could think of is somehow getting the path to the flutter used by flutter command (fvm), and then running it from home

Links to log readers below
https://github.com/flutter/flutter/blob/9d971fada15eb6570a54b65bbd13bdef4a4a16d5/packages/flutter_tools/lib/src/ios/devices.dart#L1154
https://github.com/flutter/flutter/blob/9d971fada15eb6570a54b65bbd13bdef4a4a16d5/packages/flutter_tools/lib/src/android/android_device.dart#L790

Reply:
Okey right, it seems unlikely. Then there is yet another risk that someone will only have fvm flutter on their machine. Then `flutter` command will fail as unrecognised.

Reply:
the installation guide of `fvm` recommends to have a globally installed flutter instance
https://fvm.app/documentation/getting-started/installation#recommendation
so I guess it's theoretically possible but probably not very common.

Reply:
Okey, I'm good with that